### PR TITLE
feat(patient): patient history tab

### LINKED ISSUE #2255: Create Patient History tab
## 🚀 Feature Proposal
There should be a patient history tab available in the patient profile. 

This history tab should show in chronological order events that have happened to the patient. 

To begin with this will be: 
- labs
- appointments

# Technical Notes
- Should use the `Callout` component
- will need some sort of special query that gets all entities that can be seen in the history timeline

### PR DESCRIPTION
**Pull request by VictorJuez**
Fixes #2255 - Create Patient History tab

**Changes proposed in this pull request:**

- Created a patient history tab available in the patient profile.
- Show in chronological order the events (Labs and Appointments) that have happened to the patient
    - The information shown for Labs is the Lab Type
    - The information shown for Appointments is the Appointment Reason
- Added tests for all the code created

### CODE REVIEW

**Comment by jackcmeyer** [comment]
@VictorJuez before we can review this, can  you please  fix the merge conflicts and make the pull requests checks pass. Thanks!

File: src/__tests__/patients/history/HistoryTab.test.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  })
+
+  describe('patient history list', () => {
+    it('should render patient history', async () => {
+      const { wrapper } = await setup()
+
+      expect(wrapper.exists(HistoryTable)).toBeTruthy()
+    })
+  })
+})
```
Comment (blestab) [inline src/__tests__/patients/history/HistoryTab.test.tsx:61]: May you add a newline at the of the file

File: src/patients/history/mappers/HistoryRecordsMapper.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const mapAppointments = (appointments: Appointment[] | undefined): PatientHistoryRecord[] => {
+  if (!appointments) return []
+
+  return appointments.map((appointment) => ({
+    date: new Date(appointment.startDateTime),
+    type: HistoryRecordType.APPOINTMENT,
+    info: appointment.type,
+    recordId: appointment.id,
+  }))
+}
```
Comment (blestab) [inline src/patients/history/mappers/HistoryRecordsMapper.tsx:39]: May you add a newline at the end of the file

File: src/patients/history/HistoryTable.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  const history = useHistory()
+
+  const { data: labs } = usePatientLabs(patientId)
+  const { data: appointments } = usePatientsAppointments(patientId)
+
+  const historyElements = mapHistoryRecords(labs, appointments)
+
+  return (
+    <Table
+      getID={(row) => row.id}
```
Comment (blestab) [inline src/patients/history/HistoryTable.tsx:27]: This should be recordId not id?

**Review by blestab — CHANGES_REQUESTED** [review]
I am unable to verify this functionality due to a typescript error:

```
TypeScript error in /workspace/hospitalrun-frontend/src/patients/history/HistoryTable.tsx(27,27):
Property 'id' does not exist on type 'PatientHistoryRecord'.  TS2339

    25 |   return (
    26 |     <Table
  > 27 |       getID={(row) => row.id}
       |                           ^
```

**Pull request closed without merging**